perceval/backends/mediawiki.py

```diff
@@ -0,0 +1,517 @@
+    MediaWiki site. To initialize this class the
+    URL must be provided.
+
+    :param url: MediaWiki url
+    :param cache: cache object to store raw data
+    :param origin: identifier of the repository; when `None` or an
+        empty string are given, it will be set to `url` value
+    """
+    version = '0.1.0'
+
+    def __init__(self, url, cache=None, origin=None):
+        origin = origin if origin else url
+
+        super().__init__(origin, cache=cache)
+        self.url = url
+        self.client = MediaWikiClient(url)
+
+    @metadata
+    def fetch(self, from_date=DEFAULT_DATETIME):
+        """Fetch the pages from the backend url.
+
+        The method retrieves, from a MediaWiki url, the
+        wiki pages.
+
+        :returns: a generator of pages
+        """
+
+        self._purge_cache_queue()
+
+        if from_date == DEFAULT_DATETIME:
+            from_date = None
+        else:
+            from_date = datetime_to_utc(from_date).replace(tzinfo=None)
+
+        mediawiki_version = self.client.get_version()
+        logger.info("MediaWiki version: %s", mediawiki_version)
+
+        if (mediawiki_version[0] == 1 and mediawiki_version[1] >= 27) or \
+            mediawiki_version[0] > 1:
+            fetcher = self.__fetch_1_27(from_date)
+        else:
+            fetcher = self.__fetch_pre1_27(from_date)
+
+        for page_reviews in fetcher:
+            yield page_reviews
+
+
+    @metadata
+    def fetch_from_cache(self):
+        """Fetch the pages from the cache.
+
+        :returns: a generator of pages
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = self.cache.retrieve()
+
+        pages_dicts = ['allrevisions','allpages','recentchanges']
+
+        # Namespaces -> pages -> revisions
+        next(cache_items) # First namespaces, we don't need them
+        for items in cache_items:
+            data_json = json.loads(items)
+            for pages_dict in pages_dicts:
+                if pages_dict in data_json['query']:
+                    pages_json = data_json['query'][pages_dict]
+                    for page in pages_json:
+                        page_reviews = self.__build_page_reviews(page, json.loads(next(cache_items)))
+                        yield page_reviews
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a wiki page."""
+        return str(item['pageid'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update field from a MediaWiki item.
+
+        The timestamp is extracted from 'timestamp' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        updated = None
```

Comment:
I don't see this. Does it means there are items that don't have a `updated` value???